onnx/version_converter/convert.h

```diff
@@ -611,6 +611,7 @@ class DefaultVersionConverter : public BaseVersionConverter {
     registerAdapter(std::make_unique<CompatibleAdapter>("DequantizeLinear", OpSetID(20), OpSetID(21)));
     registerAdapter(std::make_unique<CompatibleAdapter>("QuantizeLinear", OpSetID(20), OpSetID(21)));
     registerAdapter(std::make_unique<CompatibleAdapter>("QLinearMatMul", OpSetID(20), OpSetID(21)));
+    registerAdapter(std::make_unique<CompatibleAdapter>("GroupNormalization", OpSetID(20), OpSetID(21)));
```

Comment:
From our earlier discussion, I thought we had to do some transformation for the version-adapter (i.e., it is not a CompatibleAdapter). Has something changed or am I mistaken?

Reply:
@gramalingam Yes I need to change this part.

If we use an adapter, the logic should look like this:
```
    -- Shape<start=1,end=2>() -- Div(Shape_out, num_groups) 
  /                                             \
--  Reshape(,[-1,1]) ----------------- Expand(Reshape_out,[1,Div_out]) -- Reshape(, [-1])
```
There are a few pieces required here whose syntax I haven't quite figured out here:
- how to set attribute of the Shape node
- Turn `num_groups` into an initializer
- Constructing something like `[1, Div_out]`. Is it necessary to use a Constant node for `[1]` and then a Concat node to get `[1,Div_out]`?
I'm trying to see if I can find some similar examples in other adapters.

I think another option you mentioned previously was simply for this version conversion to fail. Is that an acceptable option? What are these adapters here used for: will they be called by third-party conversion tools? If so, I wonder if this will be useful since there will probably be no model out there that currently uses the opset18 definition.

Reply:
@gramalingam I added a version adapter based on the logic above. I'm not sure if it's sufficiently covered in the integration tests. Do you have a recommended way of verifying it?